src/OmniSharp.Roslyn.CSharp/Services/Refactoring/V2/CodeFixesForProjects.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using OmniSharp.Services;
+using OmniSharp.Utilities;
+
+namespace OmniSharp.Roslyn.CSharp.Services.Refactoring.V2
+{
+    [Shared]
+    [Export(typeof(CodeFixesForProjects))]
+    public class CodeFixesForProjects
+    {
+        private readonly ConcurrentDictionary<string, IEnumerable<CodeFixProvider>> _codeFixCache = new ConcurrentDictionary<string, IEnumerable<CodeFixProvider>>();
+        private readonly IAssemblyLoader _assemblyLoader;
+
+        [ImportingConstructor]
+        public CodeFixesForProjects(IAssemblyLoader assemblyLoader)
+        {
+            _assemblyLoader = assemblyLoader;
+        }
+
+        public IEnumerable<CodeFixProvider> GetAllCodeFixesForProject(string projectId)
+        {
+            if (_codeFixCache.ContainsKey(projectId))
+                return _codeFixCache[projectId];
```

Comment:
so this was confusing. so it is not actually fix it is caching, it is caching fix provider.